### DISCUSSION TITLE
variable problem input.http

### DISCUSSION THREAD

**Discussion by lowepalma** [savonet/liquidsoap#3063] [User Support]
Can someone help me, I have this function if I write the url it works fine if I use a variable it doesn't work and if I check the value of the variable by log it is the same as the one I write:

def get_radio_source(url) =
  if !stream_connected then
    # Si no hay conexión activa, utilizar la lista de reproducción
    playlist
  else
    # Si hay conexión activa, utilizar el stream
    **input.http(stream_url)    
    #input.http("https://eu1.xxxxxxx.es:2959/stream")**

  end
end

**Comment by toots** [comment]
Hi! Do you have any of of the error when using the variable?

**Reply by lowepalma** [reply]
The value of stream_url that I get in start-connect does not reach get_radio_source(url) and the value if I log it is correct, but it does not arrive if I write the url if it works

the code: 

murlstream=input.http("https://google.com")
stream_url = ref("https://google.com")


#cargar la lista.m3u
m3u_file = "lista.m3u"


#variables de control
stream_connected = ref(false)
stream = ref(blank())


#función on_new_track
def on_new_track(m) = 
  #obtener el título, el artista, el género y la duración de la canción actual
  title = m["title"]
  artist = m["artist"]
  genre = m["genre"]
  duration = m["duration"]

  song_info = "Artista: #{artist} - Título: #{title} - Género: #{genre} - Duración: #{duration} segundos"

  system("powershell -Command \"[System.IO.File]::WriteAllText('cambio.txt', '#{song_info}', [System.Text.Encoding]::UTF8)\"")
end


def on_track(cb, s) =
  s.on_track(cb)
  s
end


playlist_raw = playlist(mode="sequential", reload_mode="watch", m3u_file)
playlist = mksafe(on_track(on_new_track, playlist_raw))


pid_file = "pid.txt" 
nombre_del_exe = "liquidsoap.exe"
system("wmic process where name='#{nombre_del_exe}' get processid /format:value | findstr /r /v /c:\"^$\" > #{pid_file}")


live = input.harbor(
  "/stream",
  user = "source",
  password = "123",
  port = 8000
)

#función para determinar la fuente de audio
def get_radio_source(url) =
  if !stream_connected then
    #si no hay conexión activa, utilizar la lista de reproducción
    playlist
  else
    #si hay conexión activa, utilizar el stream
    input.http(url)
    #input.http("https://eu1.xxxx.es:2959/stream")
  end
end

#función iniciar_conex
def iniciar_conex(args)
  #dividir args en una lista de palabras
  words = string.split(separator=' ', args)
  murlstream = list.hd(words)
  
  #cambiar la URL del stream
  if not !stream_connected then
    stream_url := murlstream
    stream_connected := true
    "Conexión iniciada correctamente."
  else
    "Error: Ya existe una conexión activa."
  end
end

def desconectar_conex(_)
  #desconectar la conexión y activar la playlist
  if !stream_connected then
    stream := blank()
    stream_connected := false
    "Conexión cerrada correctamente."
  else
    "Error: No hay conexión activa para cerrar."
  end
end



#configura el servidor telnet
set("server.telnet", true)
set("server.telnet.revdns", false)
set("server.telnet.port", 4321)
set("server.telnet.bind_addr", "127.0.0.1")

#asocia el comando "iniconex" con la función iniciar_conex
server.register("iniconex", iniciar_conex)
#asocia el comando "desconex" con la función desconectar_conex
server.register("desconex", desconectar_conex)


#función para determinar si la entrada en vivo está activa
def live_connected() =
  source.is_ready(live)
end

#configurar la salida Icecast utilizando fallback
radio = switch(track_sensitive=false, [
  ({source.is_ready(live)}, live),
  ({!stream_connected}, get_radio_source(!stream_url)),
  ({true}, playlist)
])

output.icecast(
    %ffmpeg(format="mp3", %audio(codec="libmp3lame", b="128k")),
    host = "eu1.lhdserver.es",
    port = 2960,
    password = "xisco",
    mount = "stream",
    mksafe(radio)
)
